File: Leetcode 1801-1900/1855/1855.java
// Leetcode 1855: Maximum Distance Between a Pair of Values
// https://leetcode.com/problems/maximum-distance-between-a-pair-of-values/
// Solved on 25th of November, 2025
class Solution {
    /**
     * Finds the maximum distance between a pair of indices (i, j) such that i <= j, nums1[i] <= nums2[j],
     * and the distance j - i is maximized.
     * @param nums1 The first array of integers.
     * @param nums2 The second array of integers.
     * @return The maximum distance between a valid pair of indices.
     */
    public int maxDistance(int[] nums1, int[] nums2) {
        int i = 0;
        int j = 0;
        int maxDistance = 0;
        int n = nums1.length;
        int m = nums2.length;

        while (i < n && j < m) {
            if (nums1[i] > nums2[j]) {
                i++;
            } else {
                maxDistance = Math.max(maxDistance, j - i);
                j++;
            }
        }
        return maxDistance;
    }
}

File: Leetcode 1801-1900/1855/1855.py
# Leetcode 1855: Maximum Distance Between a Pair of Values
# https://leetcode.com/problems/maximum-distance-between-a-pair-of-values/
# Solved on 25th of November, 2025
class Solution:
    def maxDistance(self, nums1: list[int], nums2: list[int]) -> int:
        """
        Given two non-increasingly sorted integer arrays nums1 and nums2, return the maximum distance.
        An index pair (i, j) is a valid pair if both i <= j and nums1[i] <= nums2[j].
        The distance of an index pair (i, j) is j - i.

        :param nums1: A list of integers sorted in non-increasing order.
        :param nums2: A list of integers sorted in non-increasing order.
        :return: The maximum distance between a valid pair (i, j).
        """
        idx1 = 0
        idx2 = 0
        maxSeparation = 0
        len1 = len(nums1)
        len2 = len(nums2)

        while idx1 < len1 and idx2 < len2:
            if nums1[idx1] > nums2[idx2]:
                idx1 += 1
            else:
                currentSeparation = idx2 - idx1
                if currentSeparation > maxSeparation:
                    maxSeparation = currentSeparation
                idx2 += 1

        return maxSeparation